Installation issue: paraview+cuda+qt+python3

### Steps to reproduce the issue

When I install paraview with `cuda` and `pythnon3`, the exe `vtkpython`  failed.
```
/usr/bin/ld: lib/libvtkAcceleratorsVTKmFilters-pv5.11.a(vtkmAverageToCells.cxx.o): in function `__sti____cudaRegisterAll()':
/tmp/tmpxft_003a8074_00000000-6_vtkmAverageToCells.cudafe1.stub.c:35: undefined reference to `__cudaRegisterLinkedBinary_9f6665f5_22_vtkmAverageToCells_cxx_5ab189ac'
```

The command is
```
spack install -I -U paraview+cuda+qt+python3
```

The output of `spack spec -I -U paraview+cuda+qt+python3` is attached.
[paraview+cuda+qt+python3-spec.txt](https://github.com/spack/spack/files/9761914/paraview%2Bcuda%2Bqt%2Bpython3-spec.txt)


### Error message

<details><summary>Error message</summary><pre>
FAILED: bin/vtkpython
: && /home/z2yang/opt/spack/lib/spack/env/gcc/g++ -O2 -g -DNDEBUG -Wl,-lc  -Wl,-lc VTK/Wrapping/Python/CMakeFiles/vtkpython.dir/vtkPythonAppInit.cxx.o CMakeFiles/vtkpython.dir/cmake_device_link.o -o bin/vtkpython -L/usr/local/cuda-11.7/targets/x86_64-linux/lib/stubs   -L/usr/local/cuda-11.7/targets/x86_64-linux/lib -Wl,-rpath,/home/z2yang/opt/spack/opt/spack/linux-ubuntu22.04-zen2/gcc-11.2.0/proj-8.1.0-sxyaplda5ekjmag6y7t5nrl47oooiaoo/lib:/home/z2yang/opt/spack/opt/spack/linux-ubuntu22.04-zen2/gcc-11.2.0/libtheora-1.1.1-ps426bwcay4cllyqu4ilkpvag543vqnt/lib:/home/z2yang/opt/spack/opt/spack/linux-ubuntu22.04-zen2/gcc-11.2.0/libogg-1.3.5-sba57p35us3cghsoerxbuywqqftwzost/lib:/home/z2yang/opt/spack/opt/spack/linux-ubuntu22.04-zen2/gcc-11.2.0/libxml2-2.10.1-2vshpol4c6v5mlbgvxgdiaq7dy3vdzir/lib:/home/z2yang/opt/spack/opt/spack/linux-ubuntu22.04-zen2/gcc-11.2.0/icu4c-67.1-c7ewe7ijdkq6wtwd5fek3lakajq5qpbn/lib:/home/z2yang/opt/spack/opt/spack/linux-ubuntu22.04-zen2/gcc-11.2.0/netcdf-c-4.8.1-6v74ishg4izhvxuqdkubfbbhli5vkyrw/lib:/home/z2yang/opt/spack/opt/spack/linux-ubuntu22.04-zen2/gcc-11.2.0/jsoncpp-1.9.4-w6yib2tod6pdnv5lfvg3fokigori5bil/lib:/home/z2yang/opt/spack/opt/spack/linux-ubuntu22.04-zen2/gcc-11.2.0/hdf5-1.12.2-u7cmjjkydpbjxhtkdqkpfsytpgyjhdeq/lib:/home/z2yang/opt/spack/opt/spack/linux-ubuntu22.04-zen2/gcc-11.2.0/libx11-1.7.0-6muznwuse375442ajbimatuncdimqltl/lib:/home/z2yang/opt/spack/opt/spack/linux-ubuntu22.04-zen2/gcc-11.2.0/mesa-22.1.2-xbbcnon6whz6jk2avqe7yswkru63jvp3/lib:/home/z2yang/opt/spack/opt/spack/linux-ubuntu22.04-zen2/gcc-11.2.0/libxrender-0.9.10-exda5l5fthwielhiiljxtpaovxe3xwl5/lib:/home/z2yang/opt/spack/opt/spack/linux-ubuntu22.04-zen2/gcc-11.2.0/glew-2.2.0-rb6yddtuwhpt7vnj6cbeqqbm2jom66va/lib:/home/z2yang/opt/spack/opt/spack/linux-ubuntu22.04-zen2/gcc-11.2.0/libjpeg-turbo-2.1.3-nah7yqbaxrbr4cnsohjvjgoljwdmlu3c/lib:/home/z2yang/opt/spack/opt/spack/linux-ubuntu22.04-zen2/gcc-11.2.0/libpng-1.6.37-tacrjdyt7ldac5mds3inv7pxhg72u4bb/lib:/home/z2yang/opt/spack/opt/spack/linux-ubuntu22.04-zen2/gcc-11.2.0/zlib-1.2.12-2ujwj4a64tabxejmzahaomaldrr5dmnq/lib:/home/z2yang/opt/spack/opt/spack/linux-ubuntu22.04-zen2/gcc-11.2.0/libtiff-4.4.0-grknuppljzjzblrkvo74tt2prikwmxit/lib:/home/z2yang/opt/spack/opt/spack/linux-ubuntu22.04-zen2/gcc-11.2.0/freetype-2.11.1-r6pgueo6vwocyjqiywo2z4olizqt4xed/lib:/home/z2yang/opt/spack/opt/spack/linux-ubuntu22.04-zen2/gcc-11.2.0/expat-2.4.8-j2r2ooee6vvfm7uhjwdt6awcpejkppk6/lib:/home/z2yang/opt/spack/opt/spack/linux-ubuntu22.04-zen2/gcc-11.2.0/lz4-1.9.3-yhuiiwsxelq7oevyeyeht32lgc67nvqc/lib:/home/z2yang/opt/spack/opt/spack/linux-ubuntu22.04-zen2/gcc-11.2.0/xz-5.2.5-orcksoaevfyzqy7kuxc66ialqzgrwns7/lib:/home/z2yang/opt/spack/opt/spack/linux-ubuntu22.04-zen2/gcc-11.2.0/openmpi-4.1.4-lxjlif2jfrsyvsall5nvj6xxltv2qnzp/lib:/home/z2yang/opt/spack/opt/spack/linux-ubuntu22.04-zen2/gcc-11.2.0/python-3.9.13-b3r5vkv222ccjmksol4g25qlzz5rkv4n/lib:/home/z2yang/opt/spack/opt/spack/linux-ubuntu22.04-zen2/gcc-11.2.0/pugixml-1.11.4-i65ndtvxuvt36ar5oniaune2jahyu7ja/lib  lib/libvtkWrappingPythonCore3.9-pv5.11.a  lib/libvtkPythonInterpreter-pv5.11.a  lib/libvtksys-pv5.11.a  lib/libvtkCommonCorePython.a  lib/libvtkWebCorePython.a  lib/libvtkWebCore-pv5.11.a  lib/libvtkCommonMathPython.a  lib/libvtkCommonTransformsPython.a  lib/libvtkCommonDataModelPython.a  lib/libvtkCommonExecutionModelPython.a  lib/libvtkIOCorePython.a  lib/libvtkImagingCorePython.a  lib/libvtkIOImagePython.a  lib/libvtkIOXMLParserPython.a  lib/libvtkIOXMLPython.a  lib/libvtkCommonMiscPython.a  lib/libvtkFiltersCorePython.a  lib/libvtkRenderingCorePython.a  lib/libvtkRenderingContext2DPython.a  lib/libvtkRenderingFreeTypePython.a  lib/libvtkRenderingSceneGraphPython.a  lib/libvtkRenderingVtkJSPython.a  lib/libvtkIOExportPython.a  lib/libvtkWebGLExporterPython.a  lib/libvtkWebGLExporter-pv5.11.a  lib/libvtkFiltersGeneralPython.a  lib/libvtkFiltersSourcesPython.a  lib/libvtkInteractionWidgetsPython.a  lib/libvtkViewsCorePython.a  lib/libvtkViewsContext2DPython.a  lib/libvtkViewsContext2D-pv5.11.a  lib/libvtkViewsCore-pv5.11.a  lib/libvtkInteractionWidgets-pv5.11.a  lib/libvtkTestingRenderingPython.a  lib/libvtkTestingRendering-pv5.11.a  lib/libvtkRenderingVolumePython.a  lib/libvtkImagingMathPython.a  lib/libvtkRenderingHyperTreeGridPython.a  lib/libvtkRenderingUIPython.a  lib/libvtkRenderingOpenGL2Python.a  lib/libvtkRenderingVolumeOpenGL2Python.a  lib/libvtkRenderingVolumeAMRPython.a  lib/libvtkRenderingVolumeAMR-pv5.11.a  lib/libvtkRenderingVolumeOpenGL2-pv5.11.a  lib/libvtkRenderingVolume-pv5.11.a  lib/libvtkImagingMath-pv5.11.a  lib/libvtkRenderingParallelPython.a  lib/libvtkRenderingParallel-pv5.11.a  lib/libvtkRenderingMatplotlibPython.a  lib/libvtkRenderingMatplotlib-pv5.11.a  lib/libvtkPythonInterpreter-pv5.11.a  lib/libvtkRenderingLabelPython.a  lib/libvtkRenderingLabel-pv5.11.a  lib/libvtkRenderingLICOpenGL2Python.a  lib/libvtkRenderingLICOpenGL2-pv5.11.a  lib/libvtkRenderingContextOpenGL2Python.a  lib/libvtkRenderingContextOpenGL2-pv5.11.a  lib/libvtkParallelMPI4PyPython.a  lib/libvtkParallelMPI4Py-pv5.11.a  lib/libvtkImagingFourierPython.a  lib/libvtkImagingFourier-pv5.11.a  lib/libvtkIOLegacyPython.a  lib/libvtkIOXdmf2Python.a  lib/libvtkIOXdmf2-pv5.11.a  lib/libvtkxdmf2-pv5.11.a  lib/libvtkIOVeraOutPython.a  lib/libvtkIOVeraOut-pv5.11.a  lib/libvtkIOVPICPython.a  lib/libvtkIOVPIC-pv5.11.a  lib/libvtkvpic-pv5.11.a  lib/libvtkIOTecplotTablePython.a  lib/libvtkIOTecplotTable-pv5.11.a  lib/libvtkIOTRUCHASPython.a  lib/libvtkIOTRUCHAS-pv5.11.a  lib/libvtkIOSegYPython.a  lib/libvtkIOSegY-pv5.11.a  lib/libvtkIOParallelXMLPython.a  lib/libvtkIOParallelXML-pv5.11.a  lib/libvtkIONetCDFPython.a  lib/libvtkIOParallelNetCDFPython.a  lib/libvtkIOParallelNetCDF-pv5.11.a  lib/libvtkIONetCDF-pv5.11.a  /home/z2yang/opt/spack/opt/spack/linux-ubuntu22.04-zen2/gcc-11.2.0/proj-8.1.0-sxyaplda5ekjmag6y7t5nrl47oooiaoo/lib/libproj.so  lib/libvtkIOLSDynaPython.a  lib/libvtkIOParallelLSDynaPython.a  lib/libvtkIOParallelLSDyna-pv5.11.a  lib/libvtkIOLSDyna-pv5.11.a  lib/libvtkIOExodusPython.a  lib/libvtkIOParallelExodusPython.a  lib/libvtkIOParallelExodus-pv5.11.a  lib/libvtkIOExodus-pv5.11.a  lib/libvtkIOPLYPython.a  lib/libvtkIOPLY-pv5.11.a  lib/libvtkIOPIOPython.a  lib/libvtkIOPIO-pv5.11.a  lib/libvtkIOMoviePython.a  lib/libvtkIOOggTheoraPython.a  lib/libvtkIOOggTheora-pv5.11.a  lib/libvtkIOMovie-pv5.11.a  /home/z2yang/opt/spack/opt/spack/linux-ubuntu22.04-zen2/gcc-11.2.0/libtheora-1.1.1-ps426bwcay4cllyqu4ilkpvag543vqnt/lib/libtheora.so  /home/z2yang/opt/spack/opt/spack/linux-ubuntu22.04-zen2/gcc-11.2.0/libtheora-1.1.1-ps426bwcay4cllyqu4ilkpvag543vqnt/lib/libtheoradec.so  /home/z2yang/opt/spack/opt/spack/linux-ubuntu22.04-zen2/gcc-11.2.0/libtheora-1.1.1-ps426bwcay4cllyqu4ilkpvag543vqnt/lib/libtheoraenc.so  /home/z2yang/opt/spack/opt/spack/linux-ubuntu22.04-zen2/gcc-11.2.0/libogg-1.3.5-sba57p35us3cghsoerxbuywqqftwzost/lib/libogg.so  lib/libvtkIOOMFPython.a  lib/libvtkIOOMF-pv5.11.a  lib/libvtkIOGeometryPython.a  lib/libvtkIOParallelPython.a  lib/libvtkIOParallel-pv5.11.a  lib/libvtkIOMPIImagePython.a  lib/libvtkIOMPIImage-pv5.11.a  lib/libvtkIOInfovisPython.a  lib/libvtkIOInfovis-pv5.11.a  /home/z2yang/opt/spack/opt/spack/linux-ubuntu22.04-zen2/gcc-11.2.0/libxml2-2.10.1-2vshpol4c6v5mlbgvxgdiaq7dy3vdzir/lib/libxml2.so   /home/z2yang/opt/spack/opt/spack/linux-ubuntu22.04-zen2/gcc-11.2.0/icu4c-67.1-c7ewe7ijdkq6wtwd5fek3lakajq5qpbn/lib/libicuuc.so  lib/libvtkCommonColorPython.a  lib/libvtkImagingSourcesPython.a  lib/libvtkInfovisCorePython.a  lib/libvtkIOImportPython.a  lib/libvtkIOImport-pv5.11.a  lib/libvtkParallelCorePython.a  lib/libvtkIOIOSSPython.a  lib/libvtkIOIOSS-pv5.11.a  lib/libvtkioss-pv5.11.a  lib/libvtkexodusII-pv5.11.a  /home/z2yang/opt/spack/opt/spack/linux-ubuntu22.04-zen2/gcc-11.2.0/netcdf-c-4.8.1-6v74ishg4izhvxuqdkubfbbhli5vkyrw/lib/libnetcdf.so  lib/libvtkIOH5partPython.a  lib/libvtkIOH5part-pv5.11.a  lib/libvtkh5part-pv5.11.a  lib/libvtkIOH5RagePython.a  lib/libvtkIOH5Rage-pv5.11.a  lib/libvtkRenderingGL2PSOpenGL2Python.a  lib/libvtkIOExportGL2PSPython.a  lib/libvtkIOExportGL2PS-pv5.11.a  lib/libvtkIOExport-pv5.11.a  lib/libvtkRenderingVtkJS-pv5.11.a  lib/libvtkRenderingSceneGraph-pv5.11.a  /home/z2yang/opt/spack/opt/spack/linux-ubuntu22.04-zen2/gcc-11.2.0/jsoncpp-1.9.4-w6yib2tod6pdnv5lfvg3fokigori5bil/lib/libjsoncpp.so  lib/libvtkIOGeometry-pv5.11.a  lib/libvtklibharu-pv5.11.a  lib/libvtkRenderingGL2PSOpenGL2-pv5.11.a  lib/libvtkgl2ps-pv5.11.a  lib/libvtkIOEnSightPython.a  lib/libvtkIOEnSight-pv5.11.a  lib/libvtkIOCityGMLPython.a  lib/libvtkIOCityGML-pv5.11.a  lib/libvtkIOChemistryPython.a  lib/libvtkIOChemistry-pv5.11.a  lib/libvtkIOCONVERGECFDPython.a  lib/libvtkIOCONVERGECFD-pv5.11.a  lib/libvtkIOHDFPython.a  lib/libvtkIOHDF-pv5.11.a  lib/libvtkIOCGNSReaderPython.a  lib/libvtkIOCGNSReader-pv5.11.a  lib/libvtkcgns-pv5.11.a  lib/libvtkIOAsynchronousPython.a  lib/libvtkIOAsynchronous-pv5.11.a  lib/libvtkIOAMRPython.a  lib/libvtkIOAMR-pv5.11.a  /home/z2yang/opt/spack/opt/spack/linux-ubuntu22.04-zen2/gcc-11.2.0/hdf5-1.12.2-u7cmjjkydpbjxhtkdqkpfsytpgyjhdeq/lib/libhdf5.so.200.2.0  /home/z2yang/opt/spack/opt/spack/linux-ubuntu22.04-zen2/gcc-11.2.0/hdf5-1.12.2-u7cmjjkydpbjxhtkdqkpfsytpgyjhdeq/lib/libhdf5_hl.so.200.1.0  lib/libvtkImagingColorPython.a  lib/libvtkRenderingAnnotationPython.a  lib/libvtkRenderingAnnotation-pv5.11.a  lib/libvtkImagingColor-pv5.11.a  lib/libvtkInteractionStylePython.a  lib/libvtkInteractionStyle-pv5.11.a  lib/libvtkImagingHybridPython.a  lib/libvtkImagingHybrid-pv5.11.a  lib/libvtkFiltersPythonPython.a  lib/libvtkFiltersPython-pv5.11.a  lib/libvtkFiltersProgrammablePython.a  lib/libvtkFiltersProgrammable-pv5.11.a  lib/libvtkFiltersModelingPython.a  lib/libvtkFiltersPointsPython.a  lib/libvtkFiltersPoints-pv5.11.a  lib/libvtkFiltersVerdictPython.a  lib/libvtkFiltersParallelVerdictPython.a  lib/libvtkFiltersParallelVerdict-pv5.11.a  lib/libvtkFiltersVerdict-pv5.11.a  lib/libvtkverdict-pv5.11.a  lib/libvtkFiltersStatisticsPython.a  lib/libvtkFiltersParallelStatisticsPython.a  lib/libvtkFiltersParallelStatistics-pv5.11.a  lib/libvtkCommonSystemPython.a  lib/libvtkFiltersGeometryPython.a  lib/libvtkFiltersExtractionPython.a  lib/libvtkFiltersHybridPython.a  lib/libvtkFiltersHyperTreePython.a  lib/libvtkFiltersTexturePython.a  lib/libvtkFiltersParallelPython.a  lib/libvtkParallelMPIPython.a  lib/libvtkFiltersParallelGeometryPython.a  lib/libvtkFiltersParallelGeometry-pv5.11.a  lib/libvtkCommonComputationalGeometryPython.a  lib/libvtkFiltersFlowPathsPython.a  lib/libvtkFiltersParallelFlowPathsPython.a  lib/libvtkFiltersParallelFlowPaths-pv5.11.a  lib/libvtkFiltersFlowPaths-pv5.11.a  lib/libvtkFiltersParallelDIY2Python.a  lib/libvtkFiltersParallelDIY2-pv5.11.a  lib/libvtkFiltersImagingPython.a  lib/libvtkFiltersImaging-pv5.11.a  lib/libvtkImagingGeneralPython.a  lib/libvtkImagingGeneral-pv5.11.a  lib/libvtkFiltersGenericPython.a  lib/libvtkFiltersGeneric-pv5.11.a  lib/libvtkFiltersAMRPython.a  lib/libvtkFiltersAMR-pv5.11.a  lib/libvtkFiltersParallelMPIPython.a  lib/libvtkFiltersParallelMPI-pv5.11.a  lib/libvtkFiltersParallel-pv5.11.a  lib/libvtkFiltersModeling-pv5.11.a  lib/libvtkFiltersTexture-pv5.11.a  lib/libvtkDomainsChemistryPython.a  lib/libvtkDomainsChemistryOpenGL2Python.a  lib/libvtkDomainsChemistryOpenGL2-pv5.11.a  lib/libvtkRenderingOpenGL2-pv5.11.a  lib/libvtkRenderingHyperTreeGrid-pv5.11.a  lib/libvtkFiltersHybrid-pv5.11.a  lib/libvtkFiltersHyperTree-pv5.11.a  lib/libvtkRenderingUI-pv5.11.a  /home/z2yang/opt/spack/opt/spack/linux-ubuntu22.04-zen2/gcc-11.2.0/libx11-1.7.0-6muznwuse375442ajbimatuncdimqltl/lib/libX11.so  /home/z2yang/opt/spack/opt/spack/linux-ubuntu22.04-zen2/gcc-11.2.0/mesa-22.1.2-xbbcnon6whz6jk2avqe7yswkru63jvp3/lib/libGL.so  /usr/lib/x86_64-linux-gnu/libXcursor.so  /home/z2yang/opt/spack/opt/spack/linux-ubuntu22.04-zen2/gcc-11.2.0/libxrender-0.9.10-exda5l5fthwielhiiljxtpaovxe3xwl5/lib/libXrender.so  /usr/lib/x86_64-linux-gnu/libXfixes.so  /home/z2yang/opt/spack/opt/spack/linux-ubuntu22.04-zen2/gcc-11.2.0/libx11-1.7.0-6muznwuse375442ajbimatuncdimqltl/lib/libX11.so  lib/libvtkDomainsChemistry-pv5.11.a  /home/z2yang/opt/spack/opt/spack/linux-ubuntu22.04-zen2/gcc-11.2.0/glew-2.2.0-rb6yddtuwhpt7vnj6cbeqqbm2jom66va/lib/libGLEW.so  lib/libvtkChartsCorePython.a  lib/libvtkChartsCore-pv5.11.a  lib/libvtkRenderingContext2D-pv5.11.a  lib/libvtkInfovisCore-pv5.11.a  lib/libvtkIOImage-pv5.11.a  lib/libvtkDICOMParser-pv5.11.a  /home/z2yang/opt/spack/opt/spack/linux-ubuntu22.04-zen2/gcc-11.2.0/libjpeg-turbo-2.1.3-nah7yqbaxrbr4cnsohjvjgoljwdmlu3c/lib/libjpeg.so  lib/libvtkmetaio-pv5.11.a  /home/z2yang/opt/spack/opt/spack/linux-ubuntu22.04-zen2/gcc-11.2.0/libpng-1.6.37-tacrjdyt7ldac5mds3inv7pxhg72u4bb/lib/libpng.so  /home/z2yang/opt/spack/opt/spack/linux-ubuntu22.04-zen2/gcc-11.2.0/zlib-1.2.12-2ujwj4a64tabxejmzahaomaldrr5dmnq/lib/libz.so  /home/z2yang/opt/spack/opt/spack/linux-ubuntu22.04-zen2/gcc-11.2.0/libtiff-4.4.0-grknuppljzjzblrkvo74tt2prikwmxit/lib/libtiff.so  lib/libvtkRenderingFreeType-pv5.11.a  lib/libvtkRenderingCore-pv5.11.a  lib/libvtkFiltersSources-pv5.11.a  /home/z2yang/opt/spack/opt/spack/linux-ubuntu22.04-zen2/gcc-11.2.0/freetype-2.11.1-r6pgueo6vwocyjqiywo2z4olizqt4xed/lib/libfreetype.so  lib/libvtkCommonColor-pv5.11.a  lib/libvtkImagingSources-pv5.11.a  lib/libvtkFiltersExtraction-pv5.11.a  lib/libvtkFiltersStatistics-pv5.11.a  lib/libvtkParallelDIY-pv5.11.a  lib/libvtkIOXML-pv5.11.a  lib/libvtkIOXMLParser-pv5.11.a  /home/z2yang/opt/spack/opt/spack/linux-ubuntu22.04-zen2/gcc-11.2.0/expat-2.4.8-j2r2ooee6vvfm7uhjwdt6awcpejkppk6/lib/libexpat.so  lib/libvtkParallelMPI-pv5.11.a  lib/libvtkParallelCore-pv5.11.a  lib/libvtkIOLegacy-pv5.11.a  lib/libvtkIOCore-pv5.11.a  /home/z2yang/opt/spack/opt/spack/linux-ubuntu22.04-zen2/gcc-11.2.0/double-conversion-3.1.5-mx6cek4igg4is5g26mcrterjv4zpugdh/lib/libdouble-conversion.a  /home/z2yang/opt/spack/opt/spack/linux-ubuntu22.04-zen2/gcc-11.2.0/lz4-1.9.3-yhuiiwsxelq7oevyeyeht32lgc67nvqc/lib/liblz4.so  /home/z2yang/opt/spack/opt/spack/linux-ubuntu22.04-zen2/gcc-11.2.0/xz-5.2.5-orcksoaevfyzqy7kuxc66ialqzgrwns7/lib/liblzma.so  /home/z2yang/opt/spack/opt/spack/linux-ubuntu22.04-zen2/gcc-11.2.0/zlib-1.2.12-2ujwj4a64tabxejmzahaomaldrr5dmnq/lib/libz.so  /home/z2yang/opt/spack/opt/spack/linux-ubuntu22.04-zen2/gcc-11.2.0/openmpi-4.1.4-lxjlif2jfrsyvsall5nvj6xxltv2qnzp/lib/libmpi.so  lib/libvtkAcceleratorsVTKmCorePython.a  lib/libvtkAcceleratorsVTKmDataModelPython.a  lib/libvtkAcceleratorsVTKmFiltersPython.a    lib/libvtkWrappingPythonCore3.9-pv5.11.a  /home/z2yang/opt/spack/opt/spack/linux-ubuntu22.04-zen2/gcc-11.2.0/python-3.9.13-b3r5vkv222ccjmksol4g25qlzz5rkv4n/lib/libpython3.9.so  lib/libvtkAcceleratorsVTKmFilters-pv5.11.a  lib/libvtkImagingCore-pv5.11.a  lib/libvtkFiltersGeneral-pv5.11.a  lib/libvtkCommonComputationalGeometry-pv5.11.a  lib/libvtkfmt-pv5.11.a  lib/libvtkFiltersGeometry-pv5.11.a  lib/libvtkFiltersCore-pv5.11.a  lib/libvtkCommonExecutionModel-pv5.11.a  lib/libvtkAcceleratorsVTKmDataModel-pv5.11.a  lib/libvtkCommonDataModel-pv5.11.a  lib/libvtkCommonMisc-pv5.11.a  lib/libvtkCommonTransforms-pv5.11.a  lib/libvtkCommonMath-pv5.11.a  lib/libvtkkissfft-pv5.11.a  -lm  lib/libvtkCommonSystem-pv5.11.a  /home/z2yang/opt/spack/opt/spack/linux-ubuntu22.04-zen2/gcc-11.2.0/pugixml-1.11.4-i65ndtvxuvt36ar5oniaune2jahyu7ja/lib/libpugixml.so.1.11  lib/libvtkAcceleratorsVTKmCore-pv5.11.a  lib/libvtkCommonCore-pv5.11.a  lib/libvtksys-pv5.11.a  lib/libvtkloguru-pv5.11.a  lib/libvtkm_filter_extra-pv5.11.a  lib/libvtkm_filter_connected_components-pv5.11.a  lib/libvtkm_filter_contour-pv5.11.a  lib/libvtkm_filter_density_estimate-pv5.11.a  lib/libvtkm_filter_entity_extraction-pv5.11.a  lib/libvtkm_filter_clean_grid-pv5.11.a  lib/libvtkm_filter_image_processing-pv5.11.a  lib/libvtkm_filter_field_conversion-pv5.11.a  lib/libvtkm_filter_field_transform-pv5.11.a  lib/libvtkm_filter_geometry_refinement-pv5.11.a  lib/libvtkm_filter_mesh_info-pv5.11.a  lib/libvtkm_filter_vector_analysis-pv5.11.a  lib/libvtkm_filter_zfp-pv5.11.a  lib/libvtkm_filter_core-pv5.11.a  lib/libvtkm_worklet-pv5.11.a  lib/libvtkm_cont-pv5.11.a  lib/libvtkmdiympi_nompi.a  -ldl  -lcudadevrt  -lcudart_static  -lrt  -lpthread  -ldl && :
/usr/bin/ld: lib/libvtkAcceleratorsVTKmFilters-pv5.11.a(vtkmAverageToCells.cxx.o): in function `__sti____cudaRegisterAll()':
/tmp/tmpxft_003a8074_00000000-6_vtkmAverageToCells.cudafe1.stub.c:35: undefined reference to `__cudaRegisterLinkedBinary_9f6665f5_22_vtkmAverageToCells_cxx_5ab189ac'
/usr/bin/ld: lib/libvtkAcceleratorsVTKmFilters-pv5.11.a(vtkmAverageToPoints.cxx.o): in function `__sti____cudaRegisterAll()':
/tmp/tmpxft_003a80d8_00000000-6_vtkmAverageToPoints.cudafe1.stub.c:35: undefined reference to `__cudaRegisterLinkedBinary_63d72efe_23_vtkmAverageToPoints_cxx_6442f0a3'
/usr/bin/ld: lib/libvtkAcceleratorsVTKmFilters-pv5.11.a(vtkmCleanGrid.cxx.o): in function `__sti____cudaRegisterAll()':
/tmp/tmpxft_003a80df_00000000-6_vtkmCleanGrid.cudafe1.stub.c:35: undefined reference to `__cudaRegisterLinkedBinary_3f586887_17_vtkmCleanGrid_cxx_f7912904'
/usr/bin/ld: lib/libvtkAcceleratorsVTKmFilters-pv5.11.a(vtkmClip.cxx.o): in function `__sti____cudaRegisterAll()':
/tmp/tmpxft_003a8117_00000000-6_vtkmClip.cudafe1.stub.c:35: undefined reference to `__cudaRegisterLinkedBinary_9425639d_12_vtkmClip_cxx_3ad6d47c'
/usr/bin/ld: lib/libvtkAcceleratorsVTKmFilters-pv5.11.a(vtkmContour.cxx.o): in function `__sti____cudaRegisterAll()':
/tmp/tmpxft_003a8201_00000000-6_vtkmContour.cudafe1.stub.c:51: undefined reference to `__cudaRegisterLinkedBinary_64ce6e2b_15_vtkmContour_cxx_e38f7a81'
/usr/bin/ld: lib/libvtkAcceleratorsVTKmFilters-pv5.11.a(vtkmCoordinateSystemTransform.cxx.o): in function `__sti____cudaRegisterAll()':
/tmp/tmpxft_003a8303_00000000-6_vtkmCoordinateSystemTransform.cudafe1.stub.c:35: undefined reference to `__cudaRegisterLinkedBinary_229adcef_33_vtkmCoordinateSystemTransform_cxx_7c5e13a5'

</pre></details>


### Information on your system

* **Spack:** 0.19.0.dev0 (01ede3c595e8c02ecd791b97a0a1cae3ee30255f)
* **Python:** 3.10.4
* **Platform:** linux-ubuntu22.04-zen2
* **Concretizer:** clingo


### Additional information

[spack-build-env.txt](https://github.com/spack/spack/files/9761919/spack-build-env.txt)
[spack-build-out.zip](https://github.com/spack/spack/files/9761936/spack-build-out.zip)

@chuckatkins  @danlipsa  @vicentebolea Could you please check this?

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/paraview/package.py
+++ b/var/spack/repos/builtin/packages/paraview/package.py
@@ -109,7 +109,7 @@ class Paraview(CMakePackage, CudaPackage):
     # Python 2 support dropped with 5.9.0
     conflicts("+python", when="@5.9:")
     conflicts("+python3", when="@:5.5")
-    conflicts("+shared", when="+cuda")
+    conflicts("~shared", when="+cuda")
     conflicts("+cuda", when="@5.8:5.10")
     # Legacy rendering dropped in 5.5
     # See commit: https://gitlab.kitware.com/paraview/paraview/-/commit/798d328c

